tests/models/video_llava/test_modeling_video_llava.py

```diff
@@ -0,0 +1,577 @@
+
+            for key in model_batched_output:
+                recursive_check(model_batched_output[key], model_row_output[key], model_name, key)
+
+
+@require_torch
+class VideoLlavaForConditionalGenerationIntegrationTest(unittest.TestCase):
+    def setUp(self):
+        self.processor = VideoLlavaProcessor.from_pretrained("RaushanTurganbay/video-llava-7b-hf")
+
+    def tearDown(self):
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    @slow
+    @require_bitsandbytes
+    def test_small_model_integration_test(self):
+        # Let' s make sure we test the preprocessing to replace what is used
+        model = VideoLlavaForConditionalGeneration.from_pretrained(
+            "RaushanTurganbay/video-llava-7b-hf", load_in_4bit=True
+        )
+
+        prompt = "USER: <video>Why is this video funny? ASSISTANT:"
+        video_file = hf_hub_download(
+            repo_id="raushan-testing-hf/videos-test", filename="video_demo.npy", repo_type="dataset"
+        )
+        video_file = np.load(video_file)
+        inputs = self.processor(prompt, visual_inputs=video_file, return_tensors="pt")
+
+        EXPECTED_INPUT_IDS = torch.tensor([[1,  3148, 1001, 29901, 29871, 32001, 3750, 338, 445, 4863, 2090, 1460, 29973, 319, 1799, 9047, 13566, 29901]])  # fmt: skip
+        self.assertTrue(torch.equal(inputs["input_ids"], EXPECTED_INPUT_IDS))
+
+        output = model.generate(**inputs, do_sample=False, max_new_tokens=20)
+        EXPECTED_DECODED_TEXT = "USER:  Why is this video funny? ASSISTANT: The video is funny because the baby is playing with a Wii remote while sitting on a bed"  # fmt: skip
+
+        self.assertEqual(
+            self.processor.decode(output[0], skip_special_tokens=True),
+            EXPECTED_DECODED_TEXT,
+        )
+
+    @slow
+    @require_bitsandbytes
+    def test_small_model_integration_test_mixed_inputs(self):
+        # Let' s make sure we test the preprocessing to replace what is used
+        model = VideoLlavaForConditionalGeneration.from_pretrained(
+            "RaushanTurganbay/video-llava-7b-hf", load_in_4bit=True
+        )
+
+        prompts = [
+            "USER: <image>How many cats are there in the image? ASSISTANT:",
+            "USER: <video>Why is this video funny? ASSISTANT:",
+        ]
+        video_file = hf_hub_download(
+            repo_id="raushan-testing-hf/videos-test", filename="video_demo.npy", repo_type="dataset"
+        )
+        video_file = np.load(video_file)
+        url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        image = Image.open(requests.get(url, stream=True).raw)
+
+        inputs = self.processor(prompts, visual_inputs=[image, video_file], padding=True, return_tensors="pt")
+        output = model.generate(**inputs, do_sample=False, max_new_tokens=20)
+
+        EXPECTED_DECODED_TEXT = [
+            'USER:  How many cats are there in the image? ASSISTANT: There are two cats in the image. hopefully, they are both sleeping.',
```

Comment:
'What is this generation funny: because the addition of "hopefully" makes it seem weirdly sinister' 👀